src/main/java/com/epam/javacourses2016/task13/MyGraphCreator.java

```diff
@@ -0,0 +1,116 @@
+package com.epam.javacourses2016.task13;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MyGraphCreator extends AbstractGraphCreator {
+
+    @Override
+    public MyGraph createGraph(int numberNodes) {
+        if (numberNodes <= 0) {
+            return null;
+        }
+        return new MyGraph(numberNodes);
+    }
+
+    public class MyGraph extends AbstractGraph {
+
+        Map<Integer, Node> nodes = new HashMap<>();
+
+        public MyGraph(int numberOfNodes) {
+            super(numberOfNodes);
+        }
+
+        @Override
+        public void addEdge(int first, int second) {
+
+            if (!(checkInputValue(first) && checkInputValue(second))) {
+                return;
+            }
+
+            if (nodes.size() >= NUMBER_NODES) {
+                return;
+            }
+
+            Node firstNode = nodes.get(first);
+            Node secondNode = nodes.get(second);
+
+            if (firstNode == null) {
+                firstNode = new Node(first);
+                nodes.put(first, firstNode);
+            }
+
+            if (secondNode == null) {
+                secondNode = new Node(second);
+                nodes.put(second, secondNode);
+            }
+
+            firstNode.addNode(secondNode);
+            secondNode.addNode(firstNode);
+        }
+
+        @Override
+        public void removeEdge(int first, int second) {
+            if (!(checkInputValue(first) && checkInputValue(second))) {
+                return;
+            }
+
+            Node firstNode = nodes.get(first);
+            Node secondNode = nodes.get(second);
+
+            firstNode.getConnectedNodes().remove(secondNode);
+            secondNode.getConnectedNodes().remove(firstNode);
+        }
+
+        @Override
+        public boolean isExistEdge(int first, int second) {
+            if (!(checkInputValue(first) && checkInputValue(second))) {
+                return false;
+            }
+
+            Node firstNode = nodes.get(first);
+            Node secondNode = nodes.get(second);
+
+            if (firstNode == null) {
+                return false;
+            }
+
+            if (secondNode == null) {
+                return false;
+            }
+
+            return firstNode.getConnectedNodes().contains(secondNode) &&
+                    secondNode.getConnectedNodes().contains(firstNode);
+        }
+
+        private boolean checkInputValue(int value) {
```

Comment:
Right. But if `NUMBER_NODES` is not static (constant), why it's written in capital letters?